packages/core/src/amazonq/webview/ui/quickActions/handler.ts

```diff
@@ -45,16 +47,71 @@ export class QuickActionHandler {
                 this.handleHelpCommand(tabID)
                 break
             case '/transform':
-                this.handleGumbyCommand(tabID)
+                this.handleGumbyCommand(tabID, eventId)
                 break
             case '/clear':
                 this.handleClearCommand(tabID)
                 break
         }
     }
 
-    private handleGumbyCommand(tabID: string) {
-        this.connector.transform(tabID)
+    private handleGumbyCommand(tabID: string, eventId: string | undefined) {
+        if (!this.isGumbyEnabled) {
+            return
+        }
+
+        let gumbyTabId: string | undefined = undefined
+
+        this.tabsStorage.getTabs().forEach(tab => {
+            if (tab.type === 'gumby') {
+                gumbyTabId = tab.id
+            }
+        })
+
+        if (gumbyTabId !== undefined) {
+            this.mynahUI.selectTab(gumbyTabId, eventId || '')
+            this.connector.onTabChange(gumbyTabId)
+            return
+        }
+
+        let affectedTabId: string | undefined = tabID
+        // if there is no gumby tab, open a new one
+        if (this.tabsStorage.getTab(affectedTabId)?.type !== 'unknown') {
+            affectedTabId = this.mynahUI.updateStore('', {
+                loadingChat: true,
+                promptInputDisabledState: true,
+            })
+        }
+
+        if (affectedTabId === undefined) {
+            this.mynahUI.notify({
+                content: uiComponentsTexts.noMoreTabsTooltip,
+                type: NotificationType.WARNING,
+            })
+            return
+        } else {
+            this.tabsStorage.updateTabTypeFromUnknown(affectedTabId, 'gumby')
```

Comment:
Question: I see this `gumby` string a lot. Is it typed to a string literal in all instances or could we provide a string constant or enum value for all these instances?

Reply:
Yes, it's a [literal type](https://www.typescriptlang.org/docs/handbook/literal-types.html). Like in the previous comment, I prefer enums, but I'm following an already established convention here, and the IDE will show you 'valid' values for `TabType` as you develop so there's very little difference between it and an enum beyond preference